asf_floorplan_interpreter/pipeline/predict_floorplan.py

```diff
@@ -0,0 +1,28 @@
+import os
+from collections import Counter
+from asf_floorplan_interpreter import PROJECT_DIR
+from asf_floorplan_interpreter.utils.model_utils import load_model
+
+
+def create_results_dict(img, model):
+    """Predict a given image and output a dictionary of counts for each class
+
+    Args:
+        img (str): Image directory
+        model (pytorch model): chosen model for predictions
+    """
+
+    results = model(img, save=True)
+    class_names = results[0].names
+    class_pred_count = dict(Counter(results[0].boxes.cls.tolist()))
+
+    return {class_names[k]: v for k, v in class_pred_count.items()}
+
+
+def predict_image(directory, img):
+    "Collects model and applies to chosen images"
+    model = load_model(
```

Comment:
This code looks good, and not substantially changed from what we had.

This might be me just not understanding how to use GIthub/ S3 correctly, but I read this function as needing to put in the S3 model path as the directory (unless I should have already loaded it in a previous step to my local directory)? But then the PROJECT_DIR is the GIthub route folder, so the directory becomes garbled. It might be helpful to add some instructions (maybe in the getters)? Or if this is just a standard way of working I don't know yet, let me know!

Reply:
yes - it's actually Adeola's original code!

I think but I'm not 100% sure that we need to load the model from a local location rather than S3. Either way - as you point out, this is what the function is assuming by using PROJECT_DIR - so I will add that to the docstring and variable name just so it's cleaner. 

Perhaps later we could look into whether its easy to load directly from S3 (I had a little go but it wasn't that easy)